requests/sessions.py

```diff
@@ -633,7 +633,8 @@ def send(self, request, **kwargs):
         kwargs.setdefault('stream', self.stream)
         kwargs.setdefault('verify', self.verify)
         kwargs.setdefault('cert', self.cert)
-        kwargs.setdefault('proxies', self.rebuild_proxies(request, self.proxies))
```

Comment:
@sigmavirus24 this change is needed because without it we are calling to re-build proxies from the environment even when proxies have been set/provided. 

In one use test case I have setup - the time to "retrieve" a cached connection (using the cache control adapter) 10,000 times goes from 9.741 seconds and 25212240 function calls to 9902240 function calls & 6.169 seconds after applying this change.

Reply:
After this change - 
```         9902240 function calls (9882186 primitive calls) in 6.169 seconds

   Ordered by: internal time
   List reduced from 535 to 30 due to restriction <30>

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
        1    0.326    0.326    0.326    0.326 {method 'do_handshake' of '_ssl._SSLSocket' objects}
        1    0.191    0.191    0.191    0.191 {method 'read' of '_ssl._SSLSocket' objects}
    70004    0.185    0.000    0.553    0.000 /usr/lib/python3.9/_collections_abc.py:932(update)
   890120    0.170    0.000    0.298    0.000 {built-in method builtins.isinstance}
        1    0.150    0.150    0.150    0.150 {method 'connect' of '_socket.socket' objects}
        ...
```

Before this change ->
```        25212240 function calls (25192186 primitive calls) in 9.741 seconds

   Ordered by: internal time
   List reduced from 543 to 30 due to restriction <30>

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
  1410001    0.698    0.000    1.729    0.000 /usr/lib/python3.9/os.py:674(__getitem__)
  1510005    0.598    0.000    3.054    0.000 /usr/lib/python3.9/_collections_abc.py:848(__iter__)
  2760000    0.447    0.000    0.760    0.000 /usr/lib/python3.9/os.py:758(decode)
    10000    0.406    0.000    3.401    0.000 /usr/lib/python3.9/urllib/request.py:2486(getproxies_environment)
  1410001    0.403    0.000    0.640    0.000 /usr/lib/python3.9/os.py:754(encode)
  3010069    0.335    0.000    0.335    0.000 {method 'decode' of 'bytes' objects}
  1400000    0.321    0.000    0.689    0.000 /usr/lib/python3.9/os.py:697(__iter__)
        1    0.308    0.308    0.308    0.308 {method 'do_handshake' of '_ssl._SSLSocket' objects}
  2360120    0.258    0.000    0.380    0.000 {built-in method builtins.isinstance}
    70004    0.179    0.000    0.536    0.000 /usr/lib/python3.9/_collections_abc.py:932(update)
  1760141    0.165    0.000    0.165    0.000 {method 'lower' of 'str' objects}
        1    0.164    0.164    0.164    0.164 {method 'read' of '_ssl._SSLSocket' objects}
1490043/1490041    0.163    0.000    0.163    0.000 {method 'encode' of 'str' objects}
        1    0.147    0.147    0.147    0.147 {method 'connect' of '_socket.socket' objects}
```